orchestrator/careplanservice/handle_getpatient.go

```diff
@@ -0,0 +1,146 @@
+	if err != nil {
+		return nil, err
+	}
+	if len(patients) == 0 {
+		// If there are no patients in the bundle there is no point in doing validation, return empty bundle to user
+		return &fhir.Bundle{Entry: []fhir.BundleEntry{}}, nil
+	}
+
+	// It is possible that we have patients that are part of different CarePlans. Create list of query params for each patient ID
+	params := []fhirclient.Option{}
+	BSNs := ""
+	for i, patient := range patients {
+		if i == 0 {
+			BSNs = patientBSNFromIdentifierList(patient.Identifier)
+		} else {
+			BSNs = fmt.Sprintf("%s,%s", BSNs, patientBSNFromIdentifierList(patient.Identifier))
+		}
+	}
+	params = append(params, fhirclient.QueryParam("subject-identifier", BSNs))
+	params = append(params, fhirclient.QueryParam("_include", "CarePlan:care-team"))
+
+	// Fetch all CarePlans associated with the Patient, get the CareTeams associated with the CarePlans
+	// Get the CarePlan for which the Patient is the subject, get the CareTeams associated with the CarePlan
+	var verificationBundle fhir.Bundle
+	err = s.fhirClient.Read("CarePlan", &verificationBundle, params...)
+	if err != nil {
+		return nil, err
+	}
+
+	var careTeams []fhir.CareTeam
+	err = coolfhir.ResourcesInBundle(&verificationBundle, coolfhir.EntryIsOfType("CareTeam"), &careTeams)
+	if err != nil {
+		return nil, err
+	}
+	var carePlans []fhir.CarePlan
+	err = coolfhir.ResourcesInBundle(&verificationBundle, coolfhir.EntryIsOfType("CarePlan"), &carePlans)
+	if err != nil {
+		return nil, err
+	}
+
+	patientRefs := make([]string, 0)
+
+	// Iterate through each CareTeam to see if the requester is a participant, if not, remove any patients from the bundle that are part of the CareTeam
+	for _, cp := range carePlans {
+		var ct fhir.CareTeam
+		for _, c := range careTeams {
+			if *cp.CareTeam[0].Reference == fmt.Sprintf("CareTeam/%s", *c.Id) {
+				ct = c
+				break
+			}
+		}
+		participant := coolfhir.FindMatchingParticipantInCareTeam([]fhir.CareTeam{ct}, principal.Organization.Identifier)
+		if participant != nil {
+			for _, patient := range patients {
+				if patientBSNFromIdentifierList(patient.Identifier) == patientBSNFromIdentifier(cp.Subject.Identifier) {
+					patientRefs = append(patientRefs, "Patient/"+*patient.Id)
+					break
+				}
+			}
+		}
+	}
+
+	retBundle := filterMatchingResourcesInBundle(bundle, []string{"Patient"}, patientRefs)
+
+	return &retBundle, nil
+}
+
+func patientBSNFromIdentifier(identifier *fhir.Identifier) string {
+	if identifier.System != nil && *identifier.System == "http://fhir.nl/fhir/NamingSystem/bsn" {
+		return fmt.Sprintf("%s|%s", *identifier.System, *identifier.Value)
+	}
+	return ""
+}
+
+func patientBSNFromIdentifierList(identifier []fhir.Identifier) string {
+	for _, id := range identifier {
+		if id.System != nil && *id.System == "http://fhir.nl/fhir/NamingSystem/bsn" {
+			return fmt.Sprintf("%s|%s", *id.System, *id.Value)
+		}
+	}
+	return ""
+}
```

Comment:
Do we just support BSNs, or any Patient identifier? Since it's a Dutch number (maybe ask Bram)?

Reply:
Hmm.. That's a good point. All our resources only use BSNs but let me confirm with Bram

Reply:
We should support any patient-identifier. Patient-instances of SAZ have lot's of identifiers that they might use for updates/reads (e.g. ziekenhuis-patientnummer, EPIC-FHIR-id, etc.)

Reply:
Noted, thanks. Will make this more flexible